cpu/stm32f4/startup.c

```diff
@@ -41,6 +41,11 @@ extern void kernel_init(void);
 extern void __libc_init_array(void);
 
 /**
+ * Required by g++ cross compiler
+ */
+void* __dso_handle;
```

Comment:
What's this and isn't required for other platforms as well?

And it should be `void *__dso_handle:`